src/Stack/Setup.hs

```diff
@@ -1483,6 +1484,28 @@ installMsys2Windows osKey si archiveFile archiveType _tempDir destDir = do
     menv <- mkProcessContext newEnv
     withWorkingDir (toFilePath destDir) $ withProcessContext menv
       $ proc "sh" ["--login", "-c", "true"] runProcess_
+    -- Install MSYS2 package perl. Used by the autoreconf shell script file.
+    -- perl 5.28.1-2 is broken (see
+    -- https://github.com/msys2/MSYS2-packages/issues/1611). perl 5.28.0-1
+    -- works, so downgrading to that version, in the alternative.
+    withWorkingDir (toFilePath destDir) $ withProcessContext menv
+    --  $ proc "pacman" ["-Sy", "--noconfirm", "perl"] runProcess_
+      $ proc
```

Comment:
We've had bad experience in the past with including automated installation of pacman packages, specifically with Git before. I'd rather address this with documentation and perhaps a warning from Stack when the `autoreconf` call fails on Windows.

Reply:
Ok. I’ll revise the pull request accordingly, in due course.

Reply:
Pull request revised.